File: Solutions/0482-license-key-formatting/solution.cpp
class Solution {
public:
    string licenseKeyFormatting(string s, int k) {
        string raw = "";
        for(auto ch: s) if(ch != '-') {
            if(ch >= 'a' && ch <= 'z') raw += 'A' + (ch - 'a');
            else raw += ch;
        }
        int n = raw.length();
        int off = 0;
        string ans = "";
        if(n % k > 0) {
            int m = n % k;
            while(off < m) ans += raw[off++];
            if(off < n) ans += '-';
        }
        for(int i = 0 ; i+off < n ; i++) {
            ans += raw[i+off];
            if((i+1) % k == 0 && i+off != n-1) ans += '-';
        }
        
        return ans;
    }
};

File: Solutions/0482-license-key-formatting/solution.js
/**
 * @param {string} s
 * @param {number} k
 * @return {string}
 */
var licenseKeyFormatting = function(s, k) {
    const arr = s.split("-").reverse().map(el => el.split("").reverse().join("")).join("").toUpperCase();
    const n = arr.length;
    let ans = "";
    for(let i = 0 ; i < n ; i += k) {
        for(let j = i ; j < i + k && j < n ; j++) {
            ans += arr[j];
        }
        ans += '-';
    }
    const m = ans.length;
    ans = ans.substr(0, m-1);
    return ans.split("").reverse().join("")
};

